src/LogicTasks/Semantics/Decide.hs

```diff
@@ -110,6 +110,41 @@ start = []
 
 partialGrade :: OutputMonad m =>  DecideInst -> [Int] -> LangM m
 partialGrade DecideInst{..} sol = do
+  preventWithHint (null sol)
+    (translate $ do
+      german "Lösung enthält Indizes?"
+      english "Solution contains indices?"
+    )
+    (translate $ do
+      german "Die Lösung beinhaltet mindestens einen Index."
+      english "The solution contains at least one index."
+    )
+
+  preventWithHint (solLen > tableLen)
+    (translate $ do
+      german "Lösung überschreitet nicht maximale Anzahl Werte?"
+      english "Solution does not exceed maximum possible number of values?"
+    )
+    (translate $ do
+      german $ "Lösung enthält zu viele Werte. Es " ++ gerLong ++" entfernt werden."
+      english $ "Solution contains too many values. Please remove " ++ engLong ++ " to proceed."
+    )
+
+
+  pure ()
+  where
+    tableLen = length $ readEntries $ getTable cnf
+    solLen = length sol
+    diffToTable = abs (solLen - tableLen)
+    (gerLong,engLong) = gerEng diffToTable
+    gerEng diff = if diff == 1
+        then ("muss " ++ display ++ " Wert", display ++ " value") -- no-spell-check
+        else ("müssen " ++ display ++ " Werte", display ++ " values") -- no-spell-check
+      where
+        display = show diff
+
+completeGrade :: OutputMonad m => DecideInst -> [Int] -> LangM m
+completeGrade DecideInst{..} sol = do
   preventWithHint (solLen > acLen)
     (translate $ do
       german "Lösung enthält nicht zu viele Indizes?"
```

Comment:
Die Formulierung der/als Aufforderung weiter unten verbleibend hier ("Please remove ...") macht in `completeGrade` keinen wirklichen Sinn mehr, weil wenn man dieses Feedback zu sehen bekommt, es aus Studierendensicht zu spät für Änderungen ist (da die Bearbeitungszeit abgelaufen ist).

Reply:
Analog noch weiter unten.